File: python/example_code/cloudfront/list_distributions.py
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0


# snippet-start:[python.cloudfront.example_code.list_distributions]
import boto3


def main(cloudfront_client):
    print("CloudFront distributions:\n")
    distributions = cloudfront_client.list_distributions()
    if distributions['DistributionList']['Quantity'] > 0:
        for distribution in distributions['DistributionList']['Items']:
            print(f"Domain: {distribution['DomainName']}")
            print(f"Distribution Id: {distribution['Id']}")
            print(f"Certificate Source: "
                  f"{distribution['ViewerCertificate']['CertificateSource']}")
            if distribution['ViewerCertificate']['CertificateSource'] == "acm":
                print(f"Certificate: {distribution['ViewerCertificate']['Certificate']}")
            print("")
    else:
        print("No CloudFront distributions detected.")


if __name__ == '__main__':
    main(boto3.client('cloudfront'))
# snippet-end:[python.cloudfront.example_code.list_distributions]

File: php/example_code/cloudfront/ListDistributions.php
<?php
/*
Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0
*/

// snippet-start:[cloudfront.php.listdistribution.complete]
// snippet-start:[cloudfront.php.listdistribution.import]
require 'vendor/autoload.php';

use Aws\CloudFront\CloudFrontClient; 
use Aws\Exception\AwsException;
// snippet-end:[cloudfront.php.listdistribution.import]

/* ////////////////////////////////////////////////////////////////////////////
 * Purpose: Gets information about Amazon CloudFront distributions.
 *
 * Prerequisites: At least one existing Amazon CloudFront distribution.
 * 
 * Inputs:
 * - $cloudFrontClient: An initialized AWS SDK for PHP SDK client 
 *   for CloudFront.
 * 
 * Returns: Information about existing distributions; otherwise, 
 * the error message.
 * ///////////////////////////////////////////////////////////////////////// */

// snippet-start:[cloudfront.php.listdistribution.main]
function listDistributions($cloudFrontClient)
{
    try {
        $result = $cloudFrontClient->listDistributions([]);
        return $result;
    } catch (AwsException $e) {
        exit('Error: ' . $e->getAwsErrorMessage());
    }
}

function listTheDistributions()
{
    $cloudFrontClient = new Aws\CloudFront\CloudFrontClient([
        'profile' => 'default',
        'version' => '2018-06-18',
        'region' => 'us-east-2'
    ]);

    $distributions = listDistributions($cloudFrontClient);

    if (count($distributions) == 0)
    {
        echo 'Could not find any distributions.';
    } else {
        foreach ($distributions['DistributionList']['Items'] as $distribution)
        {
            echo 'The distribution with the ID of ' . $distribution['Id'] . 
                ' has the status of ' . $distribution['Status'] . '.' . "\n";
        }
    }
}

// Uncomment the following line to run this code in an AWS account.
// listTheDistributions();
// snippet-end:[cloudfront.php.listdistribution.main]
// snippet-end:[cloudfront.php.listdistribution.complete]
// snippet-sourcedescription:[ListDistributions.php demonstrates how to list Amazon CloudFront distributions.]
// snippet-keyword:[PHP]
// snippet-sourcesyntax:[php]
// snippet-keyword:[AWS SDK for PHP v3]
// snippet-keyword:[Code Sample]
// snippet-keyword:[listDistributions]
// snippet-keyword:[Amazon CloudFront]
// snippet-service:[cloudfront]
// snippet-sourcetype:[full-example]
// snippet-sourcedate:[2020-04-24]
// snippet-sourceauthor:[pccornel (AWS)]